### PR TITLE
Add formDataName property for managing name at content-disposition

### PR DESCRIPTION
**Pull request by platosha**
Closes #121

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-upload/126)

### CODE REVIEW

**Comment by samiheikki** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[test/upload.html, line 28 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126#-KUWEJUDeB54xIGGcXUh:-KUWEJUDeB54xIGGcXUi:bu4o2db) ([raw file](https://github.com/vaadin/vaadin-upload/blob/921de5b4fc13f9f8c272c905313eab45ffaf5d50/test/upload.html#L28)):*
> ```HTML
>       });
> 
>       function withMockFormData(callback) {
> ```

What is the purpose of this empty function?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126)*

**Comment by platosha** [comment]
Review status: 4 of 5 files reviewed at latest revision, 1 unresolved discussion.

---

*[test/upload.html, line 28 at r1](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126#-KUWEJUDeB54xIGGcXUh:-KUXY5awO3ta_RYCSW4U:b-oxf5s) ([raw file](https://github.com/vaadin/vaadin-upload/blob/921de5b4fc13f9f8c272c905313eab45ffaf5d50/test/upload.html#L28)):*
<details><summary><i>Previously, samiheikki (Sami Suo-Heikki) wrote…</i></summary>
> What is the purpose of this empty function?
</details>
Oh, commited by accident. Removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126)*

**Comment by samiheikki** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-upload.html, line 314 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126#-KUpFR4hL-0TCjii_7B2:-KUpFR4hL-0TCjii_7B3:b-shcizq) ([raw file](https://github.com/vaadin/vaadin-upload/blob/14302991d3875146ce5f56c08bc01b17882f7253/vaadin-upload.html#L314)):*
> ```HTML
> 
>       /**
>        * Specifies the 'name' property at Content-Disposition
> ```

In the original PR https://github.com/vaadin/vaadin-upload/pull/121, @manolo commented this:

This description is not clear enough. I would rephrase this to something like:

```
Specifies the 'name' of the field (file) sent to the server.
This is sent as the 'name' property of the 'Content-Disposition' header of the 'multipart/form-data' request for this field.
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126)*

**Comment by samiheikki** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-upload.html, line 314 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126#-KUpFR4hL-0TCjii_7B2:-KUq9c8zD-I0eCM_jFZ5:bnlj9bu) ([raw file](https://github.com/vaadin/vaadin-upload/blob/14302991d3875146ce5f56c08bc01b17882f7253/vaadin-upload.html#L314)):*
<details><summary><i>Previously, samiheikki (Sami Suo-Heikki) wrote…</i></summary>
> In the original PR https://github.com/vaadin/vaadin-upload/pull/121, @manolo commented this:
> 
> This description is not clear enough. I would rephrase this to something like:
> 
> ```
> Specifies the 'name' of the field (file) sent to the server.
> This is sent as the 'name' property of the 'Content-Disposition' header of the 'multipart/form-data' request for this field.
> ```
</details>
ping @manolo, is this comment still relevant?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126)*

**Comment by manolo** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-upload.html, line 314 at r2](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126#-KUpFR4hL-0TCjii_7B2:-KUzYPryAlt9h4jzsQn7:b-7qahcc) ([raw file](https://github.com/vaadin/vaadin-upload/blob/14302991d3875146ce5f56c08bc01b17882f7253/vaadin-upload.html#L314)):*
<details><summary><i>Previously, samiheikki (Sami Suo-Heikki) wrote…</i></summary>
> ping @manolo, is this comment still relevant?
</details>
not a blocker. Can be reworded later.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126)*

**Comment by samiheikki** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-upload/126#-:-KUzcXLv11upR9BGJgwf:bnfp4nl)*
